### ISSUE TITLE
Wrong  GetConnectionStatus() result _on_server_disconnected 

### ISSUE THREAD

**Issue by git2013vb** [godotengine/godot#36945]
**Godot version:**
3.2 stable mono
**OS/device including version:**
Debian 10
**Issue description:**
I still have status Connected when trigger server_disconnected signal (C#)
**Steps to reproduce:**
Check NetworkedMultiplayerPeer.ConnectionStatus when server_disconnected is triggered
**Minimal reproduction project:**
NA

**Comment by nonunknown** [comment]
plz provide a minimal reproduction project, this is required to check if the bug is really related to engine or your code!

**Comment by git2013vb** [comment]
Maybe this behavior is intended? 
It is possible the status remain as "Connected" until expire  some timeout?
If yes how I can check if the server is connected during this timeout - using godot?


Currently I use this code:
```
 private void _on_Button_pressed()
    {
        Log.Printi("On Button pressed");
        if (Global.Host.GetConnectionStatus() == NetworkedMultiplayerPeer.ConnectionStatus.Connected)
        {
            EmitSignal(nameof(login_button_pressed));
        }
        else
        {
            //TODO: make a error popup
            OS.Alert("Connection lost");
        }
    }
```

And the test I'm doing is:
0) I start the client with server down
![image](https://user-images.githubusercontent.com/13235119/76412377-6c7da900-638b-11ea-96d6-d39a554c30a9.png)

at this stage the client try to connect to the server
```
    private void _on_connection_failed()
    {
        Log.Printi("Connection failed");

        _ConnectToServer(Global.SERVER_IP_ADDRESS_LOCALHOST, Global.SERVER_PORT_NUMBER);
    }
```
1) I let client and server to be connected
![image](https://user-images.githubusercontent.com/13235119/76412608-cbdbb900-638b-11ea-83d0-4487994aec93.png)

from server I have the client ID:
![image](https://user-images.githubusercontent.com/13235119/76412754-178e6280-638c-11ea-87d4-c1c4c1bab001.png)

2) I shutdown the server and immediately i trigger the button event

as result the os.alert aren't sow  because the getconnectstatus is still in connected phase

If is intended can I adjust the timeout?

Thanks

**Comment by git2013vb** [comment]
The error still exist in 3.3 stable mono

**Comment by Faless** [comment]
The status while emitting server_disconnected is indeed still `STATE_CONNECTED`, at least for the ENet multiplayer peer.

There are also some inconsistencies among various multiplayer peers (e.g. it seems websocket only emits connection_failed but never server_disconnected). So I'll need to review those before 4.0.

**Comment by git2013vb** [comment]
> The status while emitting server_disconnected is indeed still `STATE_CONNECTED`, at least for the ENet multiplayer peer.
> 
> There are also some inconsistencies among various multiplayer peers (e.g. it seems websocket only emits connection_failed but never server_disconnected). So I'll need to review those before 4.0.

Thanks for taking a look, after two years I had lost all hope :)

**Comment by Faless** [comment]
Closing as fixed in `master` (will be in `beta5`) via #67917 .
